adoc/admin-security-nginx-ingress.adoc

```diff
@@ -63,13 +81,196 @@ helm install --name nginx-ingress suse/nginx-ingress \
+nginx-ingress-default-backend   ClusterIP      10.111.140.50    <none>        80/TCP          3h58m
+----
+
+=== Create DNS entries
+
+You should configure proper DNS names in any production environment. `k8s-dashboard.com` will be the domain name we will use in the ingress resource.
+These values are only for example purposes.
+
+* **NodePort**
+
+The services will be publicly exposed on each node of the cluster at port 30443 for HTTPS. In this example, we will use a worker node with IP 10.86.14.58.
+
+----
+k8s-dashboard.com                      IN  A       10.86.14.58
+----
+
+Or add this entry to /etc/hosts
+
+----
+10.86.14.58 k8s-dashboard.com
+----
+
+* **External IPs**
+
+The services will be exposed on a specific node of the cluster, at the assigned port for HTTPS. In this example, we used the external IP 10.86.4.158.
+
+----
+k8s-dashboard.com                      IN  A       10.86.4.158
+----
+
+Or add this entry to /etc/hosts
+
+----
+10.86.4.158 k8s-dashboard.com
+----
+* **LoadBalancer**
+
+The services will be exposed on an assigned node of the cluster, at the assigned port for HTTPS. In this example, LoadBalancer provided the external IP 10.86.5.176.
+
+----
+k8s-dashboard.com                      IN  A       10.86.5.176
+----
+
+Or add this entry to /etc/hosts
+
+----
+10.86.5.176 k8s-dashboard.com
+----
+
+== Deploy Kubernetes Dashboard as an example
+
+IMPORTANT: This example uses the upstream chart for the {kube} dashboard. There is currently no officially supported
+version of the {kube} dashboard available from {suse}.
+
+. Deploy Kubernetes dashboard
++
+[source,bash]
+----
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.0.0-rc5/aio/deploy/recommended.yaml
+----
+
+. Create cluster-admin account to access the Kubernetes dashboard
++
+This will show how to create simple admin user using Service Account, grant it the admin permission then use the token to access the kubernetes dashboard.
++
+[source,bash]
+----
+kubectl create serviceaccount dashboard-admin -n kube-system
+
+kubectl create clusterrolebinding dashboard-admin \
+--clusterrole=cluster-admin \
+--serviceaccount=kube-system:dashboard-admin
+----
+
+. Create the TLS secret
++
+Please refer to <<trusted_server_certificate>> on how to sign the trusted certificate. In this example, crt and key are generated by a self-signed certificate.
++
+[source,bash]
+----
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+-keyout /tmp/dashboard-tls.key -out /tmp/dashboard-tls.crt \
+-subj "/CN=k8s-dashboard.com/O=k8s-dashboard" \
+
+kubectl create secret tls dashboard-tls \
+--key /tmp/dashboard-tls.key --cert /tmp/dashboard-tls.crt \
+-n kubernetes-dashboard
+----
+
+. Create the ingress resource
++
+We will create an ingress to access the backend service using the ingress controller.
+Create _dashboard-ingress.yaml_ with the appropriate values
++
+----
+apiVersion: extensions/v1beta1
```

Comment:
Please verify that this actually works. The APIs have changed in k8s 1.16 and from my understand this will not work on CaaSP 4.1

Reply:
Thanks @r0ckarong 
I saw https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/
```
NetworkPolicy in the extensions/v1beta1 API version is no longer served
Migrate to use the networking.k8s.io/v1 API version, available since v1.8. Existing persisted data can be retrieved/updated via the new version.
```
I will change it and verify again. Thanks for your review.

Reply:
I tested with 
```
skuba cluster status
** This is an UNTAGGED version and NOT intended for production usage. **
NAME       STATUS   ROLE     OS-IMAGE                              KERNEL-VERSION           KUBELET-VERSION   CONTAINER-RUNTIME   HAS-UPDATES   HAS-DISRUPTIVE-UPDATES   CAASP-RELEASE-VERSION
master-0   Ready    master   SUSE Linux Enterprise Server 15 SP1   4.12.14-197.29-default   v1.16.2           cri-o://1.16.1      yes           no                       <none>
worker-0   Ready    <none>   SUSI have an error with E Linux Enterprise Server 15 SP1   4.12.14-197.29-default   v1.16.2           cri-o://1.16.1      yes           no                       <none>
```

I worked with both `extensions/v1beta1` and ` networking.k8s.io/v1beta1`. But, I have an error with `networking.k8s.io/v1`. 
```
kubectl apply -f dashboard-ingress.yaml
error: unable to recognize "dashboard-ingress.yaml": no matches for kind "Ingress" in version "networking.k8s.io/v1"
```

I agree that "networking.k8s.io/v1beta1" is newest than "extensions/v1beta1".  I will change API version as you recommended over the all ingress doc.

Reply:
Is there some information how long this will stay relevant? If 1.17 changes this around again could you please add some comment `// this is a comment in adoc` that this example location needs to be updated again? If you don't know then just please update to whatever is appropriate for 1.16 which is shipping with the current version.

Reply:
I tried to find the date or K8s version. It seems moving beta to non-beta version for ingress is not yet set in the community.